DataFormats/Math/BuildFile.xml

```diff
@@ -1,6 +1,7 @@
-<use   name="DataFormats/Common"/>
-<use   name="rootmath"/>
+<use name="eigen"/>
```

Comment:
> I do not see eigen in the permitted list of external dependencies in https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCreatingNewProducts
> Is this dependency required here?
> Can we leave basic cmath/c++ methods in choleskyInversion.h and move the `invert(Eigen::DenseBase` to somewhere else?

Currently in `cms-patatrack/CMSSW_11_1_X_Patatrack` the `choleskyInversion.h` is not used in any data formats (persistent or transient), so in principle the whole file could be moved elsewhere (not sure where though).

> @makortel @smuzaffar do you expect any issues ? If no problems, please update the twiki.
> Thank you.

@Dr15Jones, can you comment if allowing persistent data format packages to depend on Eigen could give problems?

Reply:
Does eigen start its own threads? If so, then that would be a problem. My thoughts are usually it is best to keep the DataFormats dependent upon as few externals as possible to make FWLite easier to deal with.

Reply:
> Does eigen start its own threads? If so, then that would be a problem.

Not explicitly, though it can use OpenMP pragmas to parallelise some operations.

> My thoughts are usually it is best to keep the DataFormats dependent upon as few externals as possible to make FWLite easier to deal with.

The current implementation uses Eigen extensively; however these are all under `CUDADataFormats`.

Reply:
> Currently in `cms-patatrack/CMSSW_11_1_X_Patatrack` the `choleskyInversion.h` is not used in any data formats (persistent or transient), so in principle the whole file could be moved elsewhere (not sure where though).

Why do we even have free functions under `DataFormats/Math` and `DataFormats` in general ?

Reply:
Because of dependences

Reply:
eigen is candidate to replace vector and matrices in DataFormats/ Math, GeometryVector, GeometrySurfaces

Reply:
@makortel @Dr15Jones 
are you OK then with this dependence in the DataFromats in the context of the full variety of builds that we have, and more specifically FWLite?